src/v/storage/tests/storage_e2e_test.cc

```diff
@@ -3542,3 +3545,918 @@ FIXTURE_TEST(test_skipping_compaction_below_start_offset, log_builder_fixture) {
 
     b.stop().get();
 }
+
+struct batch_summary {
+    model::offset base;
+    model::offset last;
+    size_t batch_size;
+};
+struct batch_summary_accumulator {
+    ss::future<ss::stop_iteration> operator()(model::record_batch b) {
+        size_t sz = summaries->empty() ? 0 : acc_size->back();
+        batch_summary summary{
+          .base = b.base_offset(),
+          .last = b.last_offset(),
+          .batch_size = b.data().size_bytes()
+                        + model::packed_record_batch_header_size,
+        };
+        summaries->push_back(summary);
+        acc_size->push_back(sz + summary.batch_size);
+        prev_size->push_back(sz);
+        co_return ss::stop_iteration::no;
+    }
+    bool end_of_stream() const { return false; }
+
+    std::vector<batch_summary>* summaries;
+    std::vector<size_t>* acc_size;
+    std::vector<size_t>* prev_size;
+};
+
+struct batch_size_accumulator {
+    ss::future<ss::stop_iteration> operator()(model::record_batch b) {
+        auto batch_size = b.data().size_bytes()
+                          + model::packed_record_batch_header_size;
+        *size_bytes += batch_size;
+        co_return ss::stop_iteration::no;
+    }
+    bool end_of_stream() const { return false; }
+    size_t* size_bytes;
+};
+
+FIXTURE_TEST(test_offset_range_size, storage_test_fixture) {
+#ifdef NDEBUG
```

Comment:
why not run this in debug too? maybe with less "num_test_cases" if runtime is too prohibitive

Reply:
i'd expect more invariants being asserted in underlying code/iterators/etc in debug mode

Reply:
done